test/db-migrations/migrator.js

```diff
@@ -0,0 +1,101 @@
+module.exports = {
+  exists,
+  hasRun,
+  runBefore,
+  runIncluding,
+  restoreMigrations,
+};
+
+const fs = require('node:fs');
+const { execSync } = require('node:child_process');
+
+// Horrible hacks.  Without this:
+//
+// 1. production migration code needs modifying, and
+// 2. it takes 3 mins+ just to run the migrations
+
+const migrationsDir = './lib/model/migrations';
+const holdingPen = './test/db-migrations/.holding-pen';
+
+fs.mkdirSync(holdingPen, { recursive:true });
+
+restoreMigrations();
+const allMigrations = loadMigrationsList();
+moveMigrationsToHoldingPen();
+
+let lastRunIdx = -1;
+
+function runBefore(migrationName) {
+  const idx = getIndex(migrationName);
+  if(idx === 0) return;
+
+  const previousMigration = allMigrations[idx - 1];
+
+  log('previousMigration:', previousMigration);
```

Comment:
This logging doesn't feel super useful to me given that `runIncluding()` also logs.

Reply:
:+1: In keeping with the rest of the project, I'll delete all logging before merging.